src/effects/ObjectDetection.cpp

```diff
@@ -0,0 +1,285 @@
+ * world. For more information visit <http://www.openshot.org/>.
+ *
+ * OpenShot Library (libopenshot) is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * OpenShot Library (libopenshot) is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with OpenShot Library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "effects/ObjectDetection.h"
+#include "effects/Tracker.h"
+
+using namespace std;
+using namespace openshot;
+
+/// Blank constructor, useful when using Json to load the effect properties
+ObjectDetection::ObjectDetection(std::string clipObDetectDataPath)
+{
+    // Init effect properties
+	init_effect_details();
+
+    // Tries to load the tracker data from protobuf
+    LoadObjDetectdData(clipObDetectDataPath);
+}
+
+// Default constructor
+ObjectDetection::ObjectDetection()
+{
+	// Init effect properties
+	init_effect_details();
+
+}
+
+// Init effect settings
+void ObjectDetection::init_effect_details()
+{
+	/// Initialize the values of the EffectInfo struct.
+	InitEffectInfo();
+
+	/// Set the effect info
+	info.class_name = "Object Detector";
+	info.name = "Object Detector";
+	info.description = "Detect objects through the video.";
+	info.has_audio = false;
+	info.has_video = true;
+}
+
+// This method is required for all derived classes of EffectBase, and returns a
+// modified openshot::Frame object
+std::shared_ptr<Frame> ObjectDetection::GetFrame(std::shared_ptr<Frame> frame, int64_t frame_number)
+{
+    // Get the frame's image
+	cv::Mat cv_image = frame->GetImageCV();
+    std::cout<<"Frame number: "<<frame_number<<"\n\n";
+    // Check if frame isn't NULL
+    if(!cv_image.empty()){
+
+        // Check if track data exists for the requested frame
+        if (detectionsData.find(frame_number) != detectionsData.end()) {
+            float fw = cv_image.size().width;
+            float fh = cv_image.size().height;
+
+            DetectionData detections = detectionsData[frame_number];
+            for(int i = 0; i<detections.boxes.size(); i++){
+                cv::Rect_<float> bb_nrml = detections.boxes.at(i);
+                cv::Rect2d box((int)(bb_nrml.x*fw),
+                               (int)(bb_nrml.y*fh),
+                               (int)(bb_nrml.width*fw),
+                               (int)(bb_nrml.height*fh));
+                drawPred(detections.classIds.at(i), detections.confidences.at(i),
+                         box, cv_image);
+            }
+        }
+    }
```

Comment:
You can early-return here, instead.
```suggestion
    if(cv_image.empty()){
        return frame;
    }

    // Check if track data exists for the requested frame
    if (detectionsData.find(frame_number) != detectionsData.end()) {
        float fw = cv_image.size().width;
        float fh = cv_image.size().height;

        DetectionData detections = detectionsData[frame_number];
        for(int i = 0; i<detections.boxes.size(); i++){
            cv::Rect_<float> bb_nrml = detections.boxes.at(i);
            cv::Rect2d box((int)(bb_nrml.x*fw),
                           (int)(bb_nrml.y*fh),
                           (int)(bb_nrml.width*fw),
                           (int)(bb_nrml.height*fh));
           drawPred(detections.classIds.at(i), detections.confidences.at(i),
                     box, cv_image);
        }
    }
```